cmd/embedded-cluster/metadata.go

```diff
@@ -0,0 +1,148 @@
+// embedded cluster. Release metadata involves the default versions of the
+// components that are included in the release plus the default values used
+// when deploying them.
+func gatherVersionMetadata() (*types.ReleaseMetadata, error) {
+	applier := addons.NewApplier(
+		addons.WithoutPrompt(),
+		addons.OnlyDefaults(),
+		addons.Quiet(),
+	)
+
+	versions, err := applier.Versions(config.AdditionalCharts())
+	if err != nil {
+		return nil, fmt.Errorf("unable to get versions: %w", err)
+	}
+	versions["Kubernetes"] = defaults.K0sVersion
+	versions["Installer"] = defaults.Version
+	versions["Troubleshoot"] = defaults.TroubleshootVersion
+	versions["Kubectl"] = defaults.KubectlVersion
+
+	channelRelease, err := release.GetChannelRelease()
+	if err == nil && channelRelease != nil {
+		versions[defaults.BinaryName()] = channelRelease.VersionLabel
+	}
+
+	sha, err := goods.K0sBinarySHA256()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get k0s binary sha256: %w", err)
+	}
+
+	artifacts := map[string]string{
+		"kots":                        fmt.Sprintf("kots-binaries/%s.tar.gz", adminconsole.KotsVersion),
+		"operator":                    fmt.Sprintf("operator-binaries/%s.tar.gz", embeddedclusteroperator.Metadata.Version),
+		"local-artifact-mirror-image": defaults.LocalArtifactMirrorImage,
+	}
+
+	meta := types.ReleaseMetadata{
+		Versions:  versions,
+		K0sSHA:    sha,
+		Artifacts: artifacts,
+	}
+
+	chtconfig, repconfig, err := applier.GenerateHelmConfigs(
+		config.AdditionalCharts(),
+		config.AdditionalRepositories(),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("unable to apply addons: %w", err)
+	}
+
+	meta.Configs = eckinds.Helm{
+		ConcurrencyLevel: 1,
+		Charts:           chtconfig,
+		Repositories:     repconfig,
+	}
+
+	protectedFields, err := applier.ProtectedFields()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get protected fields: %w", err)
+	}
+	meta.Protected = protectedFields
+
+	// Additional builtin addons
+	builtinCharts, err := applier.GetBuiltinCharts()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get builtin charts: %w", err)
+	}
+	meta.BuiltinConfigs = builtinCharts
+
+	cfg := config.RenderK0sConfig()
+	meta.K0sImages = config.ListK0sImages(cfg)
+
+	additionalImages, err := applier.GetAdditionalImages()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get additional images: %w", err)
+	}
+	meta.K0sImages = append(meta.K0sImages, additionalImages...)
+
+	meta.Images = append([]string{}, meta.K0sImages...)
+
+	repos := append([]eckinds.Repository{}, repconfig...)
+	charts := append([]eckinds.Chart{}, chtconfig...)
+	for _, chart := range builtinCharts {
+		repos = append(repos, chart.Repositories...)
+		charts = append(charts, chart.Charts...)
+	}
+	images, err := config.ExtractImagesFromHelmExtensions(repos, charts)
+	if err != nil {
+		return nil, fmt.Errorf("unable to extract images from helm extensions: %w", err)
+	}
+	meta.Images = append(meta.Images, images...)
```

Comment:
This is the only real new code, otherwise the rest was moved. This needs to have an internet connection because it pulls charts. Ideally we embed all charts in the binary and make it truly "embedded".

Reply:
this is only needed before we release though, right?